vscode/src/ruby/asdf.ts

```diff
@@ -29,6 +29,23 @@ export class Asdf extends VersionManager {
     const scriptName =
       path.basename(vscode.env.shell) === "fish" ? "asdf.fish" : "asdf.sh";
 
+    const config = vscode.workspace.getConfiguration("rubyLsp");
+    const asdfPath = config.get<string | undefined>("rubyVersionManager.asdfExecutablePath");
+
+    if (asdfPath) {
+      const possiblePath = vscode.Uri.joinPath(
+        vscode.Uri.parse(asdfPath),
+        scriptName,
+      );
+
+      try {
+        await vscode.workspace.fs.stat(possiblePath);
+        return possiblePath;
+      } catch (error: any) {
+        // Continue looking
+      }
+    }
```

Comment:
If the setting is intended to point at the executable path, then I don't think we should try to concatenate it with the script name, nor should we continue to search.

If the path provided by the user exists, then we return that. If it doesn't, then the user misconfigured their path and we need to halt activation since we don't know where to find ASDF.

```suggestion
    if (asdfPath) {
      const configuredPath = vscode.Uri.file(asdfPath);

      try {
        await vscode.workspace.fs.stat(configuredPath);
        return configuredPath;
      } catch (error: any) {
        throw new Error(
          `ASDF executable configured as ${configuredPath}, but that file doesn't exist`,
        );      
      }
    }
```